sotodlib/io/imprinter.py

```diff
@@ -322,6 +341,21 @@ def get_book(self, bid, session=None):
         if session is None: session = self.get_session()
         return session.query(Books).filter(Books.bid == bid).first()
```

Comment:
I would use the function `.one()` instead of `.first()` for queries that are supposed to be unique. Since book id is the primary key you should never get more than one in the query, but this is another check.

`.one_or_none()` is also useful if there's a chance the book doesn't exist.